graphql/database/users/__tests__/getUserSearchEngine.test.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-env jest */
+
+import getUserSearchEngine from '../getUserSearchEngine'
+import {
+  getMockUserContext,
+  mockDate,
+  getMockUserInstance,
+} from '../../test-utils'
+import { DEFAULT_SEARCH_ENGINE } from '../../constants'
+import getWidgets from '../../widgets/getWidgets'
+import { WIDGET_TYPE_SEARCH } from '../../../../web/src/js/constants'
+import BaseWidgetModel from '../../widgets/baseWidget/BaseWidgetModel'
+import UserWidgetModel from '../../widgets/userWidget/UserWidgetModel'
+import constructFullWidget from '../../widgets/constructFullWidget'
+import Feature from '../../experiments/FeatureModel'
+import { YAHOO_SEARCH_NEW_USERS } from '../../experiments/experimentConstants'
+import getSearchEngine from '../../search/getSearchEngine'
+
+jest.mock('../../experiments/getUserFeature')
+jest.mock('../../databaseClient')
+jest.mock('../../globals/globals')
+jest.mock('../../cause/getCause')
+jest.mock('../../widgets/getWidgets')
+
+const userContext = getMockUserContext()
+const mockCurrentTime = '2017-06-22T01:13:28.000Z'
+
+beforeAll(() => {
+  mockDate.on(mockCurrentTime, {
+    mockCurrentTimeOnly: true,
+  })
+})
+
+beforeEach(() => {
+  jest.clearAllMocks()
+})
+const user = getMockUserInstance()
+
+describe('getUserSearchEngine', () => {
+  it('defaults user engine to google if no widget, not in test or not set on model', async () => {
+    const getUserFeature = require('../../experiments/getUserFeature').default
+    getUserFeature.mockResolvedValue(undefined)
+    getWidgets.mockResolvedValueOnce([])
+
+    const result = await getUserSearchEngine(userContext, user)
+    expect(result).toEqual(await getSearchEngine('Google'))
+  })
+
+  it('defaults user engine to user feature value if no widget, or not set on model', async () => {
+    const getUserFeature = require('../../experiments/getUserFeature').default
+    getUserFeature.mockResolvedValue(
+      new Feature({
+        featureName: YAHOO_SEARCH_NEW_USERS,
+        variation: 'Google',
```

Comment:
It would be a stronger test to use the non-Google option (SearchForACause) given that Google's the default here.